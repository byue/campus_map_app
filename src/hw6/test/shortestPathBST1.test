# Simple test of breadth-first search algorithm.
# Verifies smallest path can be found on a graph with longer paths,
# same length paths but with lexicographically different children, and 
# breaking ties with lexicographically least edge labels. 

CreateGraph g

AddNode g a
AddNode g b
AddNode g d
AddNode g e
AddNode g j
AddNode g z
AddNode g f

AddEdge g a b 7
AddEdge g a a 19
AddEdge g b e 5
AddEdge g b j 8
AddEdge g b d 6
AddEdge g d z 1
AddEdge g d z 2
AddEdge g j z 9
AddEdge g e f 4
AddEdge g f z 3

FindPath g a z

FindPath g b z

FindPath g z a

FindPath g a a