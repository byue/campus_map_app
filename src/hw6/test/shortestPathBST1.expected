# Simple test of breadth-first search algorithm.
# Verifies smallest path can be found on a graph with longer paths,
# same length paths but with lexicographically different children, and
# breaking ties with lexicographically least edge labels.

created graph g

added node a to g
added node b to g
added node d to g
added node e to g
added node j to g
added node z to g
added node f to g

added edge 7 from a to b in g
added edge 19 from a to a in g
added edge 5 from b to e in g
added edge 8 from b to j in g
added edge 6 from b to d in g
added edge 1 from d to z in g
added edge 2 from d to z in g
added edge 9 from j to z in g
added edge 4 from e to f in g
added edge 3 from f to z in g

path from a to z:
a to b via 7
b to d via 6
d to z via 1

path from b to z:
b to d via 6
d to z via 1

path from z to a:
no path found

path from a to a: